### PR TITLE
Property passing. Merged click handler. Poster management. Children option.

### LINKED ISSUE #3: [RFC] Passing "leftover" props to <video>
It'd be great if it was possible to add event handlers the React way and any other unthought of or future `<video>` prop.

While I'm at it, it'd be also great if one could attach custom `onClick` handler by merging with the internal `handleClick`.

### PR DESCRIPTION
**Pull request by joltmode**
**Property passing**
Basically, with this, one can pass arbitrary props down to `<video>`, including event handlers.

**Merged click handler**
Also merged `onClick` handler with default behaviour, though, I wonder if we should provide for a way to go around default altogether?

**Poster management**
- passing `true` (which is also the default) will load poster provided by Mux.
- passing `<string>` _should_ (cause I didn't test specifically, but why wouldn't it, eh?) provide for all the ways to load a video poster.
- finally, passing `false` will skip poster altogether.

**Children option**
Could provide useful to implement, for instance, custom controls, though, as of writing this, it could have been brought up the tree... matter of preference, I guess. But I'm starting to shift more towards it being an overkill here, the component becomes too knowledgable.
But hey, it's a PR for a reason!

Fixes #3.

Sorry for so much changes in one PR, didn't want to juggle around, testing locally is a little painful for now.

### CODE REVIEW

File: src/components/Player.js [hunk 1]
```diff
@@ -135,16 +142,28 @@ class SanityMuxPlayer extends Component {
...
     }
   }
 
   // eslint-disable-next-line complexity
   render() {
     const {posterUrl, isLoading, error} = this.state
-    const {assetDocument, autoload} = this.props
+    const {
+      assetDocument, autoload,
+      showControls: showControlsProp,
```
Comment (skogsmaskin) [inline src/components/Player.js]: What is `showControlsProp`?

**Comment by skogsmaskin** [comment]
Thank you @joltmode, just one issue as reviewed.

File: src/components/Player.js (hunk 1 above)
Comment (joltmode) [reply src/components/Player.js]: Further down, on lines 181-184 `showControls` is introduced with `let` and get special treatment.

In this block, though, I destructure all props that are getting handled by `SanityMuxPlayer` internally and spread the unhandled ones into `videoProps` for safe forwarding.

Since `showControls` is the name of the prop but we also introduce `showControls` on L181, upon destructuring it here, I rename it to `showControlsProp` such that it doesn't collide with the `let showControls` call later.

Comment (joltmode) [reply src/components/Player.js]: If this is too hard to follow, and I see it is and I feared it would be, probably we should go another route for filtering unhandled props.

Introducing  `omit`, for instance.  
What I like about destructure + spread, though, is that it is actually ES6's native way of `omit`.

P.S. While writing this, I noticed that I am not lifting `onClick` out, so if `onClick` was provided to `SanityMuxPlayer`, it'd override the internal handler assignment. Meaning, in current scenario, `onClick` should also get assigned to a variable that's not actually getting used in this scope... Massive pollution, more incentive to go with `omit` utility.

**Comment by joltmode** [comment]
I reworked the prop filtering to use `omit` utility instead.

Improves code readability and predictabilty by a lot. Also cleans up scope of unused variable.

Now there is an open question, do we use the lodash one (simple way) or write a smaller footprint custom one?

**Comment by skogsmaskin** [comment]
Hmm.. I'm not sure about the mixing of props here. Could it be a better idea to have a prop `videoProps` which is exactly that, props for the video element? Then all other props are for the SanityMuxPlayer and clearly defined via PropTypes.

**Comment by joltmode** [comment]
Thought of that option also.

Yes, could go with it.

I didn't go with it initially, because I kind of disliked the idea that
most of the props you can pass to `SanityMuxPlayer` are actually the ones
you'd pass to video (`assetDocument` being the sole exception?), therefore
the extra prop to pass props of same family to the same component doesn't
mesh well.
Basically, I am not a fan of the public API in `videoProps` scenario.

**Comment by joltmode** [comment]
@skogsmaskin so - extra prop `videoProps` or keep it the current way?

**Comment by skogsmaskin** [comment]
@joltmode, I've begun refactoring what first begun as abstracting and move the error handling outside the player. That kind of escalated a bit,  and I've begun working on a custom control interface as well that should be easy to style.

The way I see it, there should be two "modes" to use this module. One mode being just plug and play where you get error handling and some easy way to style the player to fit your webpage (drop in video player). The other mode would be to just facilitate the loading of the video (src, poster and any future Sanity-specific stuff). In this mode the developer should be able to take total control of how to actually present and play the video.

Here is a suggestion for relevant props, to see if they can facilitate this:

### Player spesific
* `assetDocument`
* `onError` (by passing this you do the error handling outside the player yourself)
* `theme` (or similar, indicating that you want to use the custom player control interface, either with the standard theme or your style).
* `showControls` - toggles the themed control interface on/off (requires `theme` to be set as well)
* `children` (extra stuff to render inside the video container)

### Video spesific
* `videoElement`
  * `src`
  * `poster`
  * `loop`
  * `muted`
  * `autoPlay`
  * `controls`
  * All the rest...

What do you think?

**Comment by skogsmaskin** [comment]
OK! After discussing on Slack, let's release this for now allowing for some more control in the current API, and aim for a version `0.1.0` soon, which will have a new and much better API.

**Comment by skogsmaskin** [comment]
Thanks @joltmode! Released as `sanity-mux-player@0.0.26`
